unleash-yggdrasil/src/strategy_upgrade.rs

```diff
@@ -821,4 +834,53 @@ mod tests {
         let rule = upgrade_constraint(&constraint);
         assert_eq!(rule.as_str(), "user_id contains_any [\"some\\\"thing\"]");
     }
+
+    #[test_case("gradualRolloutUserId")]
+    #[test_case("gradualRolloutSessionId")]
+    #[test_case("gradualRolloutRandom")]
+    fn gradual_rollout_strategies_with_invalid_parameters_are_false(strategy_type: &str) {
+        let strategy = Strategy {
+            name: strategy_type.into(),
+            parameters: Some(
+                vec![("percentage".into(), "nonsense-value".into())]
```

Comment:
For the not-yet-Rust folks, this actually expects a hashmap so we make a vec of two tuples and then beat it with the `collect` stick until it becomes a hashmap

Reply:
This is the least sane thing I read all week. I love it.

So if I understood correctly essentially this represents a hashmap like this:
```
{
  percentage: nonsense-value
}
```

Is this really the best way of representing this in code?

Reply:
Yessir! That's spot on. This is fairly normal for Rust, collect pretty idiomatic for this.

There's two other options: Make a mut hashmap externally and populate it then pass that. But it's no simpler than this and probably less idiomatic

The other option is to use the map_lit crate for creating map literals but that adds to compile times and I'm trying to avoid it

Reply:
TIL. Thanks for the explanation!